### PR TITLE
[vulkan] Remove constant duplication for Vulkan optimize_for_mobile

### PR DESCRIPTION
**Pull request by IvanKobzarev**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59276 [vulkan] Remove constant duplication for Vulkan optimize_for_mobile**

Differential Revision: [D28814072](https://our.internmc.facebook.com/intern/diff/D28814072)

### CODE REVIEW

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28814072).

**Comment by cccclai** [comment]
LGTM

File: torch/csrc/jit/passes/vulkan_rewrite.cpp [hunk 1]
```diff
@@ -213,6 +214,11 @@ void vulkanRemoveMutation(script::Module& module) {
   RemoveTensorMutation(graph);
 }
 
+void vulkanRunCanonicalOptimizations(script::Module& module) {
+  auto graph = module.get_method("forward").graph();
```
Comment (cccclai) [inline torch/csrc/jit/passes/vulkan_rewrite.cpp:218]: I think we supports other optimization methods for CPU. What about vulcan?

**Review by cccclai — APPROVED** [review]

File: torch/csrc/jit/passes/vulkan_rewrite.cpp (hunk 1 above)
Comment (IvanKobzarev) [reply torch/csrc/jit/passes/vulkan_rewrite.cpp:218]: Yes, we should do the same for vulkan.

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28814072).

File: torch/csrc/jit/passes/vulkan_rewrite.cpp [hunk 2]
```diff
@@ -213,6 +214,14 @@ void vulkanRemoveMutation(script::Module& module) {
...
```
Comment (cccclai) [inline torch/csrc/jit/passes/vulkan_rewrite.cpp:218]: This `graph` is not used.

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28814072).

**Pull request closed without merging**